script.module.simplecache/lib/simplecache.py

```diff
@@ -0,0 +1,283 @@
+        query = "SELECT expires, data, checksum FROM simplecache WHERE id = ?"
+        cache_data = self.execute_sql(query, (endpoint,))
+        if cache_data:
+            cache_data = cache_data.fetchone()
+            if cache_data and cache_data[0] > cur_time:
+                if not checksum or cache_data[2] == checksum:
+                    result = eval(cache_data[1])
+                    # also set result in memory cache for further access
+                    if self.enable_mem_cache:
+                        self.set_mem_cache(endpoint, checksum, cache_data[0], result)
+        return result
+
+    def set_db_cache(self, endpoint, checksum, expires, data):
+        ''' store cache data in database '''
+        query = "INSERT OR REPLACE INTO simplecache( id, expires, data, checksum) VALUES (?, ?, ?, ?)"
+        data = repr(data)
+        self.execute_sql(query, (endpoint, expires, data, checksum))
+
+    def check_cleanup(self):
+        '''check if cleanup is needed'''
+        cur_time = datetime.datetime.now()
+        lastexecuted = self.win.getProperty("simplecache.clean.lastexecuted")
+        if not lastexecuted:
+            self.win.setProperty("simplecache.clean.lastexecuted", repr(cur_time))
+        elif (eval(lastexecuted) + self.auto_clean_interval) < cur_time:
+            # cleanup needed...
+            self.do_cleanup()
+
+    def do_cleanup(self):
+        '''perform cleanup task'''
+        if self.exit or self.monitor.abortRequested():
+            return
+        self.busy_tasks.append(__name__)
+        cur_time = datetime.datetime.now()
+        cur_timestamp = self.get_timestamp(cur_time)
+        self.log_msg("Running cleanup...")
+        if self.win.getProperty("simplecachecleanbusy"):
+            return
+        self.win.setProperty("simplecachecleanbusy", "busy")
+
+        query = "SELECT id, expires FROM simplecache"
+        for cache_data in self.execute_sql(query).fetchall():
+            if self.exit or self.monitor.abortRequested():
+                return
+            # always cleanup all memory objects on each interval
+            self.win.clearProperty(cache_data[0].encode("utf-8"))
+            # clean up db cache object only if expired
+            if cache_data[1] < cur_timestamp:
+                query = 'DELETE FROM simplecache WHERE id = ?'
+                self.execute_sql(query, (cache_data[0],))
+                self.log_msg("delete from db %s" % cache_data[0])
+
+        # compact db
+        self.execute_sql("VACUUM")
+
+        # remove task from list
+        self.busy_tasks.remove(__name__)
+        self.win.setProperty("simplecache.clean.lastexecuted", repr(cur_time))
+        self.win.clearProperty("simplecachecleanbusy")
+        self.log_msg("Auto cleanup done")
+
+    def get_database(self):
+        '''get reference to our sqllite database - performs basic integrity check'''
+        dbpath = "special://userdata/addon_data/script.module.simplecache/"
```

Comment:
I'm not so sure about that. When the module is called from a service addon, that will result in deprecated warnings in the log when xbmcaddon.Addon is called with no addon id specified.
The only improvement I can make here is defining a constant with the addon ID, and use xbmcaddon.Addon(ADDONID) but that doesn't make much of a difference imo.

What do you want me to do here ?